course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/CourseEnrolmentActor.scala

```diff
@@ -398,6 +399,13 @@ class CourseEnrolmentActor @Inject()(@Named("course-batch-notification-actor") c
             val progress: Int = enrolment.getOrDefault("progress", 0.asInstanceOf[AnyRef]).asInstanceOf[Int]
             enrolment.put("status", getCompletionStatus(progress, leafNodesCount).asInstanceOf[AnyRef])
             enrolment.put("completionPercentage", getCompletionPerc(progress, leafNodesCount).asInstanceOf[AnyRef])
+            val lrc_progress: String = enrolment.get(JsonKey.LRC_PROGRESS_DETAILS).asInstanceOf[String]
+            var mappedValue: util.Map[String, AnyRef] = new util.HashMap[String, AnyRef]()
+            if (lrc_progress != null) {
+                val objectMapper = new ObjectMapper().registerModule(DefaultScalaModule)
+                mappedValue = objectMapper.readValue(lrc_progress, classOf[util.Map[String, AnyRef]])
+            }
```

Comment:
use try and get or else -- which assigns empty map...

Reply:
updated